### ISSUE TITLE
ActionView::Template::Error on Users panel with Redmine 3.1.1

### ISSUE THREAD

**Issue by r0perice** [jbox-web/redmine_git_hosting#515]
Hi,

I'm on a fresh installation of Redmine 3.1.1 with the following configuration :

Information

```
Redmine 3.1.1.stable
Default administrator account changed   True
Attachments directory writable  True
Plugin assets directory writable (./public/plugin_assets)   True
RMagick available (optional)    True
ImageMagick convert available (optional)    True

Environment:
  Redmine version                3.1.1.stable
  Ruby version                   2.0.0-p647 (2015-08-18) [x86_64-linux]
  Rails version                  4.2.4
  Environment                    production
  Database adapter               Mysql2
SCM:
  Subversion                     1.8.13
  Git                            1.9.5
  Filesystem                     
  Xitolite                       1.7.1
Redmine plugins:
  event_notifications            3.0.0
  pdf_export                     0.0.1
  redmine_agile                  1.3.11
  redmine_auto_changes_issue_status 0.1
  redmine_bbb                    0.2.0
  redmine_bootstrap_kit          0.2.3
  redmine_checklists             3.1.1
  redmine_ckeditor               1.1.2
  redmine_create_wiki_page       0.1
  redmine_dashboard              2.7.0
  redmine_dmsf                   1.5.3
  redmine_git_hosting            1.1.3
  redmine_issue_templates        0.1.0
  redmine_ldap_sync              2.0.6.g0aec5b8345
  redmine_monitoring_controlling 0.1.1
  redmine_my_page                0.1.6
  redmine_query_share            1.0.0
  redmine_timelog_timer          2.0.1
  redmine_workflow_enhancements  0.4.0
  redmine_xapian                 1.6.2
```

I have installed redmine_git_hosting who seems to works fine. But now when I try to access to 'Users' from administration panel or if I click on an user name I'm having the following problem.

My **production.log**

```
Started GET "/redmine/users" for 80.14.188.84 at 2015-09-24 16:04:18 +0000
Processing by UsersController#index as HTML
  Current user: admin (id=1)
  Rendered plugins/redmine_git_hosting/app/views/users/index.html.erb within layouts/admin (30.1ms)
Completed 500 Internal Server Error in 85ms (ActiveRecord: 7.9ms)

ActionView::Template::Error (undefined method `unpack' for nil:NilClass):
    40:   <td class="username"><%= avatar(user, :size => "14") %><%= link_to h(user.login), edit_user_path(user) %></td>
    41:   <td class="firstname"><%= h(user.firstname) %></td>
    42:   <td class="lastname"><%= h(user.lastname) %></td>
    43:   <td class="email"><%= mail_to(h(user.mail)) %></td>
    44:   <td class="tick"><%= checked_image user.admin? %></td>
    45:   <td class="created_on"><%= format_time(user.created_on) %></td>
    46:   <td class="last_login_on"><%= format_time(user.last_login_on) unless user.last_login_on.nil? %></td>
  app/controllers/users_controller.rb:58:in `block (2 levels) in index'
  app/controllers/users_controller.rb:55:in `index'
  lib/redmine/sudo_mode.rb:63:in `sudo_mode'
```

Can you help me with that please.

Regards,

Robin

**Comment by n-rodriguez** [comment]
Hi! Are you sure this issue is due to Redmine Git Hosting? Did you try without having the plugin installed? Because there's no such method (```unpack```) in the plugin code.

**Comment by r0perice** [comment]
I just try this :
- uninstall the plugin and go to users panel : everything is ok.
- try with a new install of the plugin : same problem.
- delete all plugin and reinstall bootstrap/git_hosting : same problem

Have you tried the plugin with Redmine 3.1.1 ?

Regards,
Robin

**Comment by r0perice** [comment]
Ok this is how i resolved the problem.

I installed Redmine **3.1.1** using Bitnami wich provide Rails **4.2.4** but I think the plugin doesn't works with this version of Rails so I re-install all my gems using Rails **4.2.3**. 
For thoses who are using Bitnami stack to install Redmine **3.1.1** :
- cd REDMINE_DIR
- ./use_redmine
- cd apps/redmine/htdocs
- gem uninstall rails
- gem install rails -v=4.2.3
Edit Gemfile.lock > gem "rails", "4.2.3"

Install plugin following : http://redmine-git-hosting.io/get_started/#step-2-clone-the-plugin

Now the plugin works fine.

Maybe you can work on making the plugin compatible with Rails **4.2.4**.

Regards,
Robin

**Comment by overmind88** [comment]
Same problem for me

Redmine environment:

```
Environment:
  Redmine version                3.1.1.stable
  Ruby version                   2.0.0-p384 (2014-01-12) [x86_64-linux-gnu]
  Rails version                  4.2.4
  Environment                    production
  Database adapter               Mysql2
SCM:
  Subversion                     1.8.8
  Git                            1.9.1
  Filesystem                     
  Xitolite                       1.9.1
Redmine plugins:
  once_assigned_ever_watcher     1.0.0
  redmine_bootstrap_kit          0.2.3
  redmine_code_review            0.7.0
  redmine_git_hosting            1.1.4
  redmine_wiki_extensions        0.7.0
  redmine_wiki_gchart_formula    0.0.5
```

Errors

```
Started GET "/redmine/admin/info" for 10.0.70.119 at 2015-10-03 02:05:44 +0300
Processing by AdminController#info as HTML
  Current user: sereda (id=85)
  Rendered admin/info.html.erb within layouts/admin (41.7ms)
  Rendered admin/_menu.html.erb (2.4ms)
  Rendered plugins/redmine_code_review/app/views/code_review/_html_header.html.erb (0.0ms)
  Rendered plugins/redmine_wiki_extensions/app/views/wiki_extensions/_html_header.html.erb (0.0ms)
  Rendered plugins/redmine_code_review/app/views/code_review/_body_bottom.html.erb (0.0ms)
  Rendered plugins/redmine_wiki_extensions/app/views/wiki_extensions/_body_bottom.html.erb (0.0ms)
  Rendered layouts/base.html.erb (9.0ms)
Completed 200 OK in 66ms (Views: 55.5ms | ActiveRecord: 0.3ms)
```

```
Started GET "/redmine/users" for 10.0.70.119 at 2015-10-03 02:06:47 +0300
Processing by UsersController#index as HTML
  Current user: sereda (id=85)
  Rendered plugins/redmine_git_hosting/app/views/users/index.html.erb within layouts/admin (7.2ms)
Completed 500 Internal Server Error in 136ms (ActiveRecord: 0.5ms)

ActionView::Template::Error (undefined method `unpack' for nil:NilClass):
    40:   <td class="username"><%= avatar(user, :size => "14") %><%= link_to h(user.login), edit_user_path(user) %></td>
    41:   <td class="firstname"><%= h(user.firstname) %></td>
    42:   <td class="lastname"><%= h(user.lastname) %></td>
    43:   <td class="email"><%= mail_to(h(user.mail)) %></td>
    44:   <td class="tick"><%= checked_image user.admin? %></td>
    45:   <td class="created_on"><%= format_time(user.created_on) %></td>
    46:   <td class="last_login_on"><%= format_time(user.last_login_on) unless user.last_login_on.nil? %></td>
  app/controllers/users_controller.rb:58:in `block (2 levels) in index'
  app/controllers/users_controller.rb:55:in `index'
  lib/redmine/sudo_mode.rb:63:in `sudo_mode'
```

After downgrade to rails 4.2.3 works fine

**Comment by n-rodriguez** [comment]
I can't reproduce the bug :
```
Environment:
  Redmine version                3.1.1.stable
  Ruby version                   2.1.5-p273 (2014-11-13) [x86_64-linux]
  Rails version                  4.2.4
  Environment                    development
  Database adapter               Mysql2
SCM:
  Subversion                     1.8.10
  Mercurial                      3.1.2
  Git                            2.1.4
  Filesystem                     
  Xitolite                       2.1.4
Redmine plugins:
  redmine_bootstrap_kit          0.2-devel
  redmine_git_hosting            1.1-devel
  redmine_sidekiq                2.1.0
  time_logger                    0.5.3
```

```
Started GET "/admin/info" for x.x.x.x at 2015-10-03 20:44:22 +0200
Processing by AdminController#info as HTML
  Current user: admin (id=1)
Version: ImageMagick 6.8.9-9 Q16 x86_64 2015-01-05 http://www.imagemagick.org
Copyright: Copyright (C) 1999-2014 ImageMagick Studio LLC
Features: DPC Modules OpenMP
Delegates: bzlib djvu fftw fontconfig freetype jbig jng jpeg lcms lqr ltdl lzma openexr pangocairo png tiff wmf x xml zlib

sh: 1: darcs: not found
sh: 1: cvs: not found
sh: 1: bzr: not found
  Rendered admin/info.html.erb within layouts/admin (4786.5ms)
  Rendered admin/_menu.html.erb (7.9ms)
  Rendered plugins/time_logger/app/views/time_loggers/_embed_menu.html.erb (10.2ms)
  Rendered plugins/time_logger/app/views/time_loggers/_update_menu.html.erb (12.6ms)
  Rendered layouts/base.html.erb (54.6ms)
Completed 200 OK in 5558ms (Views: 4865.6ms | ActiveRecord: 16.2ms)
```

```
Started GET "/users" for x.x.x.x at 2015-10-03 20:45:43 +0200
Processing by UsersController#index as HTML
  Current user: admin (id=1)
  Rendered plugins/redmine_git_hosting/app/views/users/index.html.erb within layouts/admin (21.1ms)
  Rendered admin/_menu.html.erb (6.3ms)
  Rendered plugins/time_logger/app/views/time_loggers/_embed_menu.html.erb (4.6ms)
  Rendered plugins/time_logger/app/views/time_loggers/_update_menu.html.erb (6.4ms)
  Rendered layouts/base.html.erb (25.4ms)
Completed 200 OK in 87ms (Views: 57.6ms | ActiveRecord: 15.8ms)
```

**Comment by n-rodriguez** [comment]
``` 
ActionView::Template::Error (undefined method `unpack' for nil:NilClass):
...
app/controllers/users_controller.rb:58:in `block (2 levels) in index
'```

It seems to come from the render method : http://www.redmine.org/projects/redmine/repository/entry/trunk/app/controllers/users_controller.rb#L58

**Comment by AdamN1** [comment]
Same problem for me
```
Started GET "/users" for 195.54.47.32 at 2015-10-04 23:39:23 +0200
Processing by UsersController#index as HTML
  Current user: m_o_d (id=1)
  Rendered plugins/redmine_git_hosting/app/views/users/index.html.erb within layouts/admin (7.0ms)
Completed 500 Internal Server Error in 16ms (ActiveRecord: 1.6ms)

ActionView::Template::Error (undefined method `unpack' for nil:NilClass):
    40:   <td class="username"><%= avatar(user, :size => "14") %><%= link_to h(user.login), edit_user_path(user) %></td>
    41:   <td class="firstname"><%= h(user.firstname) %></td>
    42:   <td class="lastname"><%= h(user.lastname) %></td>
    43:   <td class="email"><%= mail_to(h(user.mail)) %></td>
    44:   <td class="tick"><%= checked_image user.admin? %></td>
    45:   <td class="created_on"><%= format_time(user.created_on) %></td>
    46:   <td class="last_login_on"><%= format_time(user.last_login_on) unless user.last_login_on.nil? %></td>
  app/controllers/users_controller.rb:58:in `block (2 levels) in index'
  app/controllers/users_controller.rb:55:in `index'
  lib/redmine/sudo_mode.rb:63:in `sudo_mode'
```

Can i help debug?

**Comment by AdamN1** [comment]
like @r0perice wrote backport to rails 4.2.3 solve problems

**Comment by shugg** [comment]
Also running into the same issue:

```
Processing by UsersController#index as HTML
  Current user: admin (id=1)
  Rendered plugins/redmine_git_hosting/app/views/users/index.html.erb within layouts/admin (6.6ms)
Completed 500 Internal Server Error in 16ms (ActiveRecord: 2.3ms)

ActionView::Template::Error (undefined method `unpack' for nil:NilClass):
    40:   <td class="username"><%= avatar(user, :size => "14") %><%= link_to h(user.login), edit_user_path(user) %></td>
    41:   <td class="firstname"><%= h(user.firstname) %></td>
    42:   <td class="lastname"><%= h(user.lastname) %></td>
    43:   <td class="email"><%= mail_to(h(user.mail)) %></td>
    44:   <td class="tick"><%= checked_image user.admin? %></td>
    45:   <td class="created_on"><%= format_time(user.created_on) %></td>
    46:   <td class="last_login_on"><%= format_time(user.last_login_on) unless user.last_login_on.nil? %></td>
  app/controllers/users_controller.rb:58:in `block (2 levels) in index'
  app/controllers/users_controller.rb:55:in `index'
  lib/redmine/sudo_mode.rb:63:in `sudo_mode'
```

My environment:

```
Environment:
  Redmine version                3.1.1.stable
  Ruby version                   2.0.0-p598 (2014-11-13) [x86_64-linux]
  Rails version                  4.2.4
  Environment                    production
  Database adapter               Mysql2
SCM:
  Subversion                     1.7.14
  Git                            1.8.3.1
  Filesystem                     
  Xitolite                       1.8.3.1
Redmine plugins:
  redmine_bootstrap_kit          0.2.3
  redmine_git_hosting            1.1.4
```

**Comment by cdelalande** [comment]
thx @r0perice it works

**Comment by netsuo** [comment]
Seems to be a problem with Rails 4.2.4, I don't use the bitnami stack but I followed @r0perice ' advice and downgraded Rails to 4.2.3 (even without reinstalling everything) and now it works.

**Comment by Varkal** [comment]
Same issue here. Don't use the binami stack but the install instructions here : http://www.redmine.org/projects/redmine/wiki/HowTo_Install_Redmine_30x_on_Ubuntu_1404_with_Apache2_Phusion_Passenger_MySQL_Subversion_and_Git_%28Gitolite%29

Downgrade to Rails 4.2.3 and now its works

**Comment by hans42** [comment]
Hey,
i'm also struggling with this issue. Is there already a way to make it work with Rails 4.2.4?

**Comment by n-rodriguez** [comment]
> i'm also struggling with this issue. Is there already a way to make it work with Rails 4.2.4?

Well this is weird because I can't reproduce this issue : https://github.com/jbox-web/redmine_git_hosting/issues/515#issuecomment-145277190

**Comment by hans42** [comment]
At the end downgrading to Rails 4.2.3 solved the problem for me, too

**Comment by marudesign** [comment]
I'm fixed same problems.
 bitnami-rubystack with redmine module after upgrade rails 4.2.4 -> 4.2.5

My version Info 

Environment:
  Redmine version                3.1.2.stable
  Ruby version                   2.0.0-p647 (2015-08-18) [x86_64-linux]
  Rails version                  4.2.5
  Environment                    production
  Database adapter               Mysql2
SCM:
  Subversion                     1.8.13
  Git                            2.6.1
  Filesystem                     
  Xitolite                       2.6.1
Redmine plugins:
  redmine_agile                  1.3.12
  redmine_bootstrap_kit          0.2.4
  redmine_ckeditor               1.1.2
  redmine_dashboard              2.7.0
  redmine_git_hosting            1.2.0

base : Ruby Stack 2.0.0-34 (64-bit) for bitnami.com

**Comment by juanlealz** [comment]
We solved it upgrading from ruby 1.9, rails 4.2.4 to ruby 2.1.7, 4.2.4.
